FRAuth/FRAuth/SelfService/DeviceClient.swift

```diff
@@ -253,3 +212,49 @@ public class DeviceClient: DeviceRepository {
         return try JSONDecoder().decode(Session.self, from: data)
     }
 }
+
+
+/// Implementation of the `ImmutableDevice` protocol for managing devices that only support retrieval and deletion.
+@available(iOS 13.0.0, *)
+public struct ImmutableDeviceImplementation<R>: ImmutableDevice where R: Device {
+  var endpoint: String
```

Comment:
Ti my previous comment, I think that should be enough, as this is a Public struct and a devloper could add their values and set it. On that note, I suggest we add some code docs here on what values are expected for these properties